lib/samples/query_table_statistics/query_table_statistics_sample.dart

```diff
@@ -29,117 +29,190 @@ class QueryTableStatisticsSample extends StatefulWidget {
+                  // Add a map view to the widget tree and set a controller.
+                  child: ArcGISMapView(
+                    controllerProvider: () => _mapViewController,
+                    onMapViewReady: onMapViewReady,
                   ),
-                  SwitchListTile(
-                    title: const Text('Only cities greater than 5M'),
-                    value: _onlyCitiesGreaterThan5M,
-                    onChanged: (value) {
-                      setState(() => _onlyCitiesGreaterThan5M = value);
-                    },
-                  ),
-                  TextButton(
-                    onPressed: queryStatistics,
-                    child: const Text(
-                      'Get statistics',
-                    ),
-                  ),
-                ],
+                ),
+              ],
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            Visibility(
+              visible: !_ready,
+              child: SizedBox.expand(
+                child: Container(
+                  color: Colors.white30,
+                  child: const Center(child: CircularProgressIndicator()),
+                ),
               ),
             ),
-          ),
-        ],
+
+            Visibility(
+              visible: _toggleQuerySettings,
+              child: querySettings(context),
+            ),
+          ],
+        ),
       ),
+      bottomSheet: null, //querySettings(context),
     );
   }
 
+  // Display the query options to query statistics.
+  Widget querySettings(BuildContext context) {
+    return Positioned(
+      bottom: 50,
+      left: 50,
+      width: MediaQuery.of(context).size.width * 0.75,
+      child: Container(
+        color: Colors.white,
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            Row(
+              children: [
+                Checkbox(
+                  value: _onlyCitiesInCurrentExtent,
+                  onChanged: (value) {
+                    setState(() => _onlyCitiesInCurrentExtent = value!);
+                  },
+                ),
+                const Text('Only cities in current extent'),
+              ],
+            ),
+            Row(
+              children: [
+                Checkbox(
+                  value: _onlyCitiesGreaterThan5M,
+                  onChanged: (value) {
+                    setState(() => _onlyCitiesGreaterThan5M = value!);
+                  },
+                ),
+                const Text('Only cities greater than 5M'),
+              ],
+            ),
+            ElevatedButton(
+              onPressed: queryStatistics,
+              child: const Text(
+                'Get statistics',
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  // Called when the map view is ready.
+  void onMapViewReady() {
+    for (final type in StatisticType.values) {
```

Comment:
This method needs the usual sort of tutorial comments, describing what is being done.